addons/website_sale_delivery/controllers/main.py

```diff
@@ -43,8 +43,21 @@ def cart_carrier_rate_shipment(self, carrier_id, **kw):
         carrier = request.env['delivery.carrier'].sudo().browse(int(carrier_id))
         rate = carrier.rate_shipment(order)
         if rate.get('success'):
+            tax_id = carrier.product_id.taxes_id.filtered(lambda t: t.company_id.id == order.company_id.id)[0]
```

Comment:
It can work with multiple taxes.
```suggestion
            tax_id = carrier.product_id.taxes_id.filtered(lambda t: t.company_id.id == order.company_id.id)
```